### PR TITLE
fix: improve screen reader experience by fixing redundant FAB.Group item announcements.

### LINKED ISSUE #4063: Screen reader: Items in FAB group each have three focusable regions
### Current behaviour

Currently, when using a screen reader (tested with TalkBack on Android) to navigate a `FAB.Group`, each group item appears in navigation three times (twice for the label, once for the button).

### Expected behaviour

Each item should appear in navigation exactly once.

### How to reproduce?

[GitHub repository](https://github.com/personalizedrefrigerator/bug-report/tree/react-native-paper-broken-fab) or [Expo Snack](https://snack.expo.dev/@personalizedrefrigerator/react-native-paper-accessibility-bug)

See specifically [`MainScreen.tsx`](https://github.com/personalizedrefrigerator/bug-report/blob/react-native-paper-broken-fab/MainScreen.tsx).

### Preview

https://github.com/callstack/react-native-paper/assets/46334387/6004ef8d-fc31-45be-be74-4ee536d4a7ff

### Your Environment

| software                      | version
| --------------------- | -------
| ios                                | untested
| android                        | 10
| react-native                | 0.72.4
| react-native-paper    | 5.10.4
| node                            | 18.13.0
| npm or yarn                | 9.6.7 (npm)
| expo sdk                     | N/A

### PR DESCRIPTION
**Pull request by abdulbasithqb**
## Motivation
The goal is to enhance the screen reader experience by resolving redundant item announcements in the FAB.Group.

## Related Issue
Fixes #4063

## Test Plan
1. **Check FAB.Group Presence:**
   - Verify the presence of FAB.Group in FABExample.tsx.

2. **Android Testing (TalkBack):**
   - Use TalkBack on Android to navigate the FAB.Group.
   - Confirm that each item is announced only once.
   - Ensure that no new accessibility issues are introduced.

3. **iOS Testing (VoiceOver):**
   - Verify the presence and functionality of FAB.Group on iOS devices.
   - Use VoiceOver to navigate the FAB.Group.
   - Confirm that each item is announced only once.
   - Ensure that no new accessibility issues are introduced.

Remember to document any findings, changes, or considerations during the testing process. This comprehensive plan should help ensure a thorough evaluation of the accessibility improvements across different platforms and devices.

## Result

| Android  | IOS |
| ------------- | ------------- |
| <video title src="https://github.com/callstack/react-native-paper/assets/134603758/066c6c33-3064-4a7e-8d8e-80ae33955436">  | <video src="https://github.com/callstack/react-native-paper/assets/134603758/03c51cf7-71e1-406b-9628-a7029d034615">|

### CODE REVIEW

**Comment by abdulbasithqb** [comment]
**iOS changes have been incorporated to improve accessibility**
commit : [ae1c1f3](https://github.com/callstack/react-native-paper/pull/4196/commits/ae1c1f33869f574e23d9ee2a28df7e93c2648700)

https://github.com/callstack/react-native-paper/assets/134603758/4e5e18ff-1b0f-45ac-9656-fccd0ada80a1

File: src/components/FAB/FABGroup.tsx [hunk 1]
```diff
@@ -390,8 +394,8 @@ const FABGroup = ({
                         close();
                       }}
                       accessibilityHint={it.accessibilityHint}
-                      accessibilityLabel={accessibilityLabel}
-                      accessibilityRole="button"
+                      importantForAccessibility="no-hide-descendants"
+                      accessibilityElementsHidden={true}
```
Comment (lukewalczak) [inline src/components/FAB/FABGroup.tsx:398]:
```suggestion
                      accessibilityElementsHidden
```

File: src/components/FAB/FABGroup.tsx [hunk 2]
```diff
@@ -408,6 +412,8 @@ const FABGroup = ({
                     >
                       <Text
                         variant="titleMedium"
+                        importantForAccessibility="no-hide-descendants"
+                        accessibilityElementsHidden={true}
```
Comment (lukewalczak) [inline src/components/FAB/FABGroup.tsx:416]:
```suggestion
                        accessibilityElementsHidden
```

File: src/components/FAB/FABGroup.tsx [hunk 3]
```diff
@@ -429,13 +435,13 @@ const FABGroup = ({
                     isV3 && { transform: [{ translateY: translations[i] }] },
                     it.style,
                   ]}
+                  accessibilityElementsHidden={true}
```
Comment (lukewalczak) [inline src/components/FAB/FABGroup.tsx:438]:
```suggestion
                  accessibilityElementsHidden
```

**Review by lukewalczak — APPROVED** [review]

**Pull request merged**